### PR TITLE
Sd 4834 publish send email event from wf

### PR DESCRIPTION
**Pull request by bhaskarpawar123**

### CODE REVIEW

**Review by alpeshjikadra — CHANGES_REQUESTED** [review]
Needs to fix integration test data

File: src/test/resources/test-scripts/integration/insert-workflow-for-email-action-test-record-primary-email.sql [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+INSERT INTO abstract_workflow_action(id, workflow_id)
+VALUES ('a0eebc55-9c0b-4ef8-bb6d-6bb9bd380a11', 301),
+('b0eebc55-9c0b-4ef8-bb6d-6bb9bd380a11', 302);
+
+INSERT INTO email_recipient(id, email_type, entity_type, entity_id, name, email)
+VALUES ('a0eebc55-9c0b-4ef8-bb6d-6bb9bd380a14','RECORD_PRIMARY_EMAIL', 'lead', 2, 'test', 'test@123.com'),
+('a0eebc55-9c0b-4ef8-bb6d-6bb9bd380a15','RECORD_PRIMARY_EMAIL', 'lead', 3, 'test', 'test@123.com'),
+('a0eebc55-9c0b-4ef8-bb6d-6bb9bd380a16','RECORD_PRIMARY_EMAIL', 'lead', 4, 'test', 'test@123.com'),
+('a0eebc55-9c0b-4ef8-bb6d-6bb9bd380a17','RECORD_OWNER', 'user', 12, 'user1', 'user1@gmail.com');
```
Comment (alpeshjikadra) [inline src/test/resources/test-scripts/integration/insert-workflow-for-email-action-test-record-primary-email.sql:52]: Why there is value for entityId and email for this entity_type ?

**Review by alpeshjikadra — CHANGES_REQUESTED** [review]

File: src/it/java/com/kylas/sales/workflow/EmailActionIntegrationTests.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+import org.testcontainers.containers.RabbitMQContainer;
+
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+@AutoConfigureTestDatabase(replace = Replace.NONE)
+@AutoConfigureWireMock(port = 9090)
+@ContextConfiguration(initializers = {TestMqSetup.class, TestDatabaseInitializer.class})
+@TestPropertySource(properties = {"client.sales.basePath=http://localhost:9090"})
+public class EmailActionIntegrationTests {
+
+  private final String authenticationToken =
```
[the remark this answers is not in the document]
Comment (bhaskarpawar123) [reply src/it/java/com/kylas/sales/workflow/EmailActionIntegrationTests.java:68]: resolved. I am using it now..

File: src/it/java/com/kylas/sales/workflow/EmailActionIntegrationTests.java [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+  public static void tearDown() {
+    rabbitMQContainer.stop();
+  }
+
+  @Test
+  @Sql("/test-scripts/integration/insert-workflow-for-email-action-test.sql")
+  public void givenLeadCreatedEvent_withEmailAction_shouldPublishEmailActionEvent() throws IOException, InterruptedException {
+    //given
+    User aUser = UserStub.aUser(12L, 99L, true, true, true, true, true).withName("Steve");
+    given(authService.getLoggedInUser()).willReturn(aUser);
```
[the remark this answers is not in the document]
Comment (bhaskarpawar123) [reply src/it/java/com/kylas/sales/workflow/EmailActionIntegrationTests.java]: i have removed mocks from the class

File: src/test/resources/test-scripts/integration/insert-workflow-for-email-action-test-record-primary-email.sql (hunk 1 above)
Comment (bhaskarpawar123) [reply src/test/resources/test-scripts/integration/insert-workflow-for-email-action-test-record-primary-email.sql]: changed in all the places wherever it is required.

File: src/main/java/com/kylas/sales/workflow/domain/WorkflowFacade.java [hunk 4]
```diff
@@ -419,4 +427,63 @@ private static String readFromInputStream(InputStream inputStream) throws IOExce
           return user;
         });
   }
+
+  public void updateEntityDetailInEmailRecipient(EmailType emailType, EntityEvent entityEvent) {
```
[the remark this answers is not in the document]
Comment (bhaskarpawar123) [reply src/main/java/com/kylas/sales/workflow/domain/WorkflowFacade.java:431]: We need this method to update the lead/contact information, when lead/contact updated if it is available in any of to, cc, or bcc

**Review by alpeshjikadra — APPROVED** [review]

**Review by alpeshjikadra — APPROVED** [review]

**Pull request merged**